README.md

```diff
@@ -1,3 +1,90 @@
 # repo-template
 
 **This is not an official Google product.**
+
+This template is for creating an AOD (access on demand) repository to get access
+to GCP resources on demand.
+
+## Make an AOD Request
+
+**Make sure you have completed
+[prerequisite steps](https://github.com/abcxyz/aod-template/blob/main/README.md#prerequisites)
+to set up this repo properly.**
+
+1.  Open a PR
+
+    -   A title and description to provide enough justification. E.g. "AOD
+        request: debug customer issue X".
+    -   To request IAM permissions on org/folder/project level, add an
+        `iam.yaml` file in the repo root.
+    -   See an example of this file, see example in
+        [example-iam.yaml](https://github.com/abcxyz/aod-template/example-iam.yaml).
+    -   Optionally use predefined duration labels on the PR to specify the IAM
+        permission expiration. Otherwise a 2h default duration will be used.
+    -   (Later) To request on-demand `gcloud` commands, add an `gcloud.yaml`
+        file in the repo root.
+
+2.  Checks
+
+    -   All AOD PRs will fail the check "do_not_merge" to prevent accidental
+        merge.
+    -   The "Validate Request" check will report errors if there are problems in
+        your `iam.yaml` or `gcloud.yaml`.
+
+3.  Getting approval
+
+    -   Ask one of the repo code owners to approve the PR
+    -   Upon approval, another GitHub workflow will run to grant you the IAM
+        permissions or run the on-demand `gcloud` commands
+
+4.  Close the PR after you no longer need the access. The PR will automatically
+    be closed after X hours (TODO #8).
+
+## How AOD works(TODO #7)
+
+## Prerequisites
+
+The admin of your GCP project/folder/org to complete the steps below.
+
+1.  Create an AOD repository using this template, only copy main branch is
+    required.
+
+2.  Setup
```

Comment:
Set up [Workload Identity Federation](https://cloud.google.com/iam/docs/workload-identity-federation) by following guide [here](https://github.com/google-github-actions/auth#setting-up-workload-identity-federation)?

Reply:
For now yes, we don't have a better guide for WIF yet.

Reply:
I mean you can change the sentence to `Set up [Workload Identity Federation](https://cloud.google.com/iam/docs/workload-identity-federation) by following guide [here](https://github.com/google-github-actions/auth#setting-up-workload-identity-federation)`